integration-tests/test_ui.py

```diff
@@ -42,7 +42,12 @@ async def local_hub_local_binder(request):
                 break
         except requests.exceptions.ConnectionError:
             pass
-        await asyncio.sleep(1)
+        # We use a increased linear time between requests
```

Comment:
Rather than using a linear increase we can just do `for i in range(60)`, (summing the linear increases means this takes up to 45s, so we can round it off to 1 minute). I'd keep `await asyncio.sleep(1)` since although it's not the cause of the bug this is an async function, and using async.sleep will avoid future problems.